src/nvim/highlight_group.c

```diff
@@ -799,31 +799,37 @@ int lookup_color(const int idx, const bool foreground, TriState *const boldp)
 void set_hl_group(int id, HlAttrs attrs, Dict(highlight) *dict, int link_id)
 {
   int idx = id - 1;  // Index is ID minus one.
-
   bool is_default = attrs.rgb_ae_attr & HL_DEFAULT;
-
-  // Return if "default" was used and the group already has settings
-  if (is_default && hl_has_settings(idx, true)) {
```

Comment:
isn't it enough to just check if `force` is set here and then override this check right here? I don't think making the code more convoluted below is needed or desired.

Reply:
before tried but test case failed.

Reply:
these will failed if move before link set the `default link` can't set .  test case from issue 23023 reproduce step
 
```lua
    meths.set_hl(0, 'Bar', {link ='Boolean', default = true })
    eq('Bar            xxx guifg=#123456',
      exec_capture('highlight Bar'))
    command('hi clear')
    eq('Bar            xxx links to Boolean',
      exec_capture('highlight Bar'))
```